tests/modules/programs/mbsync/mbsync-expected.conf

```
@@ -56,8 +56,8 @@ IMAPAccount hm@example.com
 CertificateFile /etc/ssl/certs/ca-certificates.crt
 Host imap.example.com
 PassCmd password-command
-SSLType IMAPS
-SSLVersions TLSv1.3 TLSv1.2
+TLSType IMAPS
+TLSVersions TLSv1.3 TLSv1.2
```

```suggestion
TLSVersions +1.3 +1.2
```

Sorry about the drive-by comment, but i just got bitten by this change in isync/mbsync. The `SSLVersions` and `TLSVersions` config keys don't take the same arguments, it's not a simple rename.

I have found references here https://sourceforge.net/projects/isync/files/isync/1.5.0/ and here https://isync.sourceforge.io/mbsync.html but frustratingly i haven't found actual documentation for permitted arguments.  By trial and error i managed to make things work with the `+` as shown above.

Disclosure/disclaimer is that i'm using isync/mbsync on macOS via Homebrew – i am just making a comment here before this change wastes other people's time more than necessary. I found this PR because i was looking for working examples.

---

That's a good catch! Thanks! I will have to mark `SSLVersions` as completely deprecated and inform the user about `TLSVersions` along with the `+`/`-` change, I guess.

>  ...but frustratingly i haven't found actual documentation for permitted arguments.

I found the source file that implements this parsing (`src/drv_imap.c:3788`). They also support a `-` symbol to disable a particular version. I suspect this was an undocumented change to support isync 1.5.0's default-disabling of TLS 1.0/1.1.

I will document this in a release note. But because `SSLVersions`/`TLSVersions` is part of `extraConfig`, I doubt I will be able to automatically catch this misconfiguration.

---

> isync 1.5.0's default-disabling of TLS 1.0/1.1

This is the only reason I used `SSLVersions` so I could delete the line. (I probably won't, though, in case there are future changes.)

---

Is this something that needs to be resolved before merging?

---

Thanks for the clarification @dbaynard.